### PR TITLE
Metadata fix

### PR DESCRIPTION
**Pull request by fiona-gladwin**
- Change data type of Image ID from uint to int.
- Add support to create empty instance of metadata class when augmentation nodes are enabled and copy all info, without copying contents of labels, bbox and mask.
- Remove resize of the metadata from master graph and move to clone API

### CODE REVIEW

File: rocAL/rocAL/include/meta_data/meta_data.h [hunk 1]
```diff
@@ -292,10 +292,14 @@ class LabelBatch : public MetaDataBatch
     }
     std::shared_ptr<MetaDataBatch> clone(bool copy_contents) override
     {
-        if(copy_contents)
+        if(copy_contents) {
```
Comment (rrawther) [inline rocAL/rocAL/include/meta_data/meta_data.h:295]: I think both branches are making copies of the labelsBatch. Why do we need to treat them separate?

File: rocAL/rocAL/include/meta_data/meta_data.h [hunk 2]
```diff
@@ -362,10 +367,14 @@ class BoundingBoxBatch: public LabelBatch
...
             return std::make_shared<BoundingBoxBatch>(*this);
```
Comment (rrawther) [inline rocAL/rocAL/include/meta_data/meta_data.h]: same applies to all other similar instances

**Review by rrawther — CHANGES_REQUESTED** [review]
please address review comments

File: rocAL/rocAL/include/meta_data/meta_data.h (hunk 1 above)
Comment (fiona-gladwin) [reply rocAL/rocAL/include/meta_data/meta_data.h:295]: The if condition - Creates copy of the complete metadata batch along with labels, bbox and all the info
The else condition - Creates a new instance of the metadata batch, it copies only the info and does not copy the labels, bbox and mask as it is expected to be filled in by the metanodes

**Comment by LakshmiKumar23** [comment]
> please address review comments

@rrawther This looks good to merge. Please confirm

**Review by LakshmiKumar23 — APPROVED** [review]

File: rocAL/rocAL/include/meta_data/meta_data.h (hunk 1 above)
Comment (rrawther) [reply rocAL/rocAL/include/meta_data/meta_data.h:295]: I am not sure if this is the right practice. When is the clone used?

Comment (fiona-gladwin) [reply rocAL/rocAL/include/meta_data/meta_data.h:295]: Clone is used in the output routine
1. We fill the metadata batch struct with metadata of respective images using look up
2. We need to create a copy of the metadata batch to be stored into the ring buffer
3. To create the copy we use clone.

If metanodes are enabled - We are creating a new instance of the metadata batch, it copies only the info and does not copy the labels, bbox and mask as it is expected to be filled in by the metanodes
else - We need to clone the entire metadata along with labels, bbox and all the info

**Comment by fiona-gladwin** [comment]
@rrawther as discussed added comments in the clone API for readability

**Review by rrawther — APPROVED** [review]

**Pull request merged**